### PR TITLE
Update in IT-Pixel part of the Phase2Digitizer

### PR DESCRIPTION
**Pull request by suchandradutta**
#### PR description:
Two fixes  for the Phase2 IT (InnerTracker) pixel  (InnerTracker) digitizer algorithm and in configuration :
 - cross-talk implementation: added a tolerance of ~0.5 micron to select 25x100 um pixels by the pixel size as in the latest geometries the pixel dimension is 25x100.115 μm (NB: by default cross-talk is off so this change has no effect)

-  Parametrization of charge diffusion: SigmaCoeff=0 reverts the parametrization of charge diffusion to the predictions found in literature (accidentally charge diffusion in IT pixel was parametrized with a function derived in beam tests of OT modules).

#### PR validation:
Basic test procedure including runTheMatrix test works (followed (https://cms-sw.github.io/PRWorkflow.html) 
> 34 33 32 26 14 4 1 1 1 tests passed, 0 0 0 0 0 0 0 0 0 failed

Before submitting your pull requests, make sure you followed this checklist:
- verify that the PR is really intended for the chosen branch
- verify that changes follow [CMS Naming, Coding, And Style Rules](http://cms-sw.github.io/cms_coding_rules.html)
- verify that the PR passes the basic test procedure suggested in the [CMSSW PR instructions](https://cms-sw.github.io/PRWorkflow.html)

@emiglior

### CODE REVIEW

**Comment by emiglior** [comment]
Please assign this PR as "upgrade"
@pwittich this is of interest for you

**Comment by mmusich** [comment]
> Please assign this PR as "upgrade"

This PR is already upgrade-assigned, though automatic testing and notifications won't happen until the code quality issues reported at  https://github.com/cms-sw/cmssw/pull/28100#issuecomment-537079998 will be resolved.

**Comment by fabiocos** [comment]
please test

File: SimTracker/SiPhase2Digitizer/plugins/PixelDigitizerAlgorithm.cc [hunk 1]
```diff
@@ -134,9 +134,12 @@ void PixelDigitizerAlgorithm::add_cross_talk(const Phase2TrackerGeomDetUnit* pix
   // cross-talk calculation valid for the case of 25x100 pixels
   const float pitch_first = 0.0025;
   const float pitch_second = 0.0100;
+  const double pitch_tolerance(
+      0.0005);  // 0.5 um tolerance when comparing the pitch to accommodate the small changes in different TK geometrie (temporary fix)
```
Comment (kpedro88) [inline SimTracker/SiPhase2Digitizer/plugins/PixelDigitizerAlgorithm.cc]: it's better to put the comment on a separate line so the code format doesn't split the line in an awkward way like this

**Review by suchandradutta — COMMENTED** [review]
Updated separating the comment line from the code

**Comment by mmusich** [comment]
@suchandradutta @emiglior 
this PR is stuck in code-check-rejected state since two weeks. Is it still desired ? (I would naively say yes).

**Comment by fabiocos** [comment]
please test

**Comment by kpedro88** [comment]
That was my understanding.

**Comment by civanch** [comment]
+1

**Comment by fabiocos** [comment]
+1

**Pull request merged**